Kudu.Core/Jobs/ContinuousJobsManager.cs

```diff
@@ -32,7 +33,13 @@ public ContinuousJobsManager(ITraceFactory traceFactory, IEnvironment environmen
 
         public override IEnumerable<ContinuousJob> ListJobs()
         {
-            return ListJobsInternal();
+            var cache = ContinuousJobCache;
+            if (cache == null)
+            {
+                cache = ListJobsInternal();
+                ContinuousJobCache = cache;
+            }
+            return cache;
```

Comment:
oh I take that back. That's to handle race conditions with the change notification isn't it.